polaris/ocean/ice_shelf/ssh_forward.py

```diff
@@ -0,0 +1,175 @@
+from polaris.ocean.model import OceanModelStep, get_time_interval_string
+
+
+class SshForward(OceanModelStep):
+    """
+    A step for performing forward ocean component runs as part of ssh
+    adjustment.
+
+    Attributes
+    ----------
+    resolution : float
+        The minimum resolution in km used to determine the time step
+
+    package : Package
+        The package name or module object that contains ``namelist``
+
+    yaml_filename : str, optional
+        the yaml filename used for ssh_forward steps
+
+    yaml_replacements : Dict, optional
+        key, string combinations for templated replacements in the yaml
+        file
+    """
+    def __init__(self, component, min_resolution, mesh, init,
```

Comment:
The `mesh` is just used as a way to get a graph filename, so it would allow for better generality if we have the graph file as the input.

Similarly, `init` is only used for the previous output file, so it would be more general to take that filename.

This would then allow the input graph file to come from a different step than the initial condition (not currently possible) and the output from the original `init` step could be called `init.nc` or something like that instead of `output.nc` as required here.